x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/components/post_install_cloud_formation_modal.tsx

```diff
@@ -41,9 +45,19 @@ export const PostInstallCloudFormationModal: React.FunctionComponent<{
 
   const cloudFormationProps = getCloudFormationPropsFromPackagePolicy(packagePolicy);
 
+  const fleetServerHostId = agentPolicy?.fleet_server_host_id;
+  const { data } = useGetFleetServerHosts();
+
+  const fleetServerHosts = useMemo(() => data?.items ?? [], [data?.items]);
+
+  const selectedHost = fleetServerHosts.find((host) => host.id === fleetServerHostId);
+
+  const fleetServerUrl = selectedHost?.host_urls?.[0];
```

Comment:
considering adding comment explaining why we take the first one, as this might create problems in the future. problems that for right now we choose to accept

Reply:
FYI: https://github.com/elastic/kibana/issues/166155